### ISSUE TITLE
VHost metadata in logs and attributes

### ISSUE THREAD

**Issue by guydc** [envoyproxy/envoy#34900]
*Title*: VHost metadata in logs and attributes

*Description*:
Envoy resource metadata (e.g. route, cluster and listener metadata) is typically supported in the following:
- [X] Access Logging: Metadata can be logged using the metadata formatter extension (`%METADATA(*:key)%`) : https://github.com/envoyproxy/envoy/pull/34958
- [X] Attributes: metadata can be accessed through an attribute (`xds.*_metadata`) https://github.com/envoyproxy/envoy/pull/37650

VHost metadata was introduced in #30175. VHost metadata should also support these capabilities. 

*Relevant Links*:
- https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/formatter/metadata/v3/metadata.proto.html#envoy-v3-api-msg-extensions-formatter-metadata-v3-metadata
- https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes

**Comment by mathetake** [comment]
I just opened the PR for this: https://github.com/envoyproxy/envoy/pull/34958

**Comment by mathetake** [comment]
merged and can be closed 😉

**Comment by guydc** [comment]
Hi @mathetake ! 

Does the PR also cover: 
> Attributes: metadata can be accessed through an attribute (xds.*_metadata)

**Comment by mathetake** [comment]
the PR is not limited to access log, but wherever the METADATA( substitution is used

**Comment by mathetake** [comment]
yeah, I think the second one is not implemented (though I am not knowledgeable on that part)

**Comment by guydc** [comment]
thanks @wbpcode for implementing the attributes part in https://github.com/envoyproxy/envoy/pull/37650.
